UI/obs-app.cpp

```diff
@@ -1528,9 +1532,42 @@ const char *OBSApp::GetCurrentLog() const
 	return currentLogFile.c_str();
 }
 
+#ifdef __APPLE__
+static DIR *mac_crashlog_directory =
+	opendir(strcat(getenv("HOME"), "/Library/Logs/DiagnosticReports"));
+#endif
+
 const char *OBSApp::GetLastCrashLog() const
 {
+#if !defined(__APPLE__)
 	return lastCrashLogFile.c_str();
+#else
+	struct dirent *dir;
+	char *report = nullptr;
+	if (mac_crashlog_directory) {
+		while ((dir = readdir(mac_crashlog_directory)) != NULL) {
+			/** Filter out all non-OBS crash logs */
+			if (strncmp(dir->d_name, "obs", 3) == 0) {
+				if (!report ||
+				    strcmp(dir->d_name, report) > 0) {
+					report = dir->d_name;
+				}
+			}
+		}
+		rewinddir(mac_crashlog_directory);
+	} else {
+		blog(LOG_WARNING,
+		     "Couldn't upload a crash log since the directory seems to be missing.");
+	}
+	if (report) {
+		return report;
```

Comment:
Returning the address of a statically allocated `dirent` member is unsafe as it may be overwritten by another thread at any time.